userweb/app/modules/commonView/headerCtrl.js

```diff
@@ -46,7 +46,23 @@ angular
             $scope.userAgent = localStorage.getItem('userAgent');
             $scope.ipRegistered = localStorage.getItem('ipRegistered');
         }
-        
+
+        var gsQRCode = localStorage.getItem('gsQRCode');
+
+        if (gsQRCode == 1) {
+            var terminalName = localStorage.getItem('terminalName');
+            var terminalPassword = localStorage.getItem('terminalPassword');
+            var extensionNumber = localStorage.getItem('extensionNumber');
+            var companyName = localStorage.getItem('companyName');
+            var companyDomain = localStorage.getItem('companyDomain');
+            var userName = localStorage.getItem('userName');
+
+            $scope.gsQRCode = true;
+            $scope.QRCode = '<?xml version="1.0" encoding="utf-8"?><AccountConfig version="1"><Account><RegisterServer>' + companyDomain + '</RegisterServer><UserID>' + terminalName + '</UserID><AuthID>' + terminalName + '</AuthID><AuthPass>' + terminalPassword + '</AuthPass><AccountName>' + extensionNumber + ' ' + companyName + '</AccountName><DisplayName>' + userName + '</DisplayName><Voicemail>*93</Voicemail></Account></AccountConfig>';
```

Comment:
This could improve readability with a multine string, what about:

```            
$scope.QRCode = '<?xml version="1.0" encoding="utf-8"?>'
            + '<AccountConfig version="1"><Account>'
                + '<RegisterServer>' + companyDomain + '</RegisterServer>'
                + '<UserID>' + terminalName + '</UserID>'
                + '<AuthID>' + terminalName + '</AuthID>'
                + '<AuthPass>' + terminalPassword + '</AuthPass>'
                + '<AccountName>' + extensionNumber + ' ' + companyName + '</AccountName>'
                + '<DisplayName>' + userName + '</DisplayName>'
                + '<Voicemail>*93</Voicemail>'
            + '</Account></AccountConfig>';
```